### PR TITLE
RENO-2175: MenuGroup Component

### PR DESCRIPTION
**Pull request by nypl-wluisi**
[Jira Ticket](https://jira.nypl.org/browse/RENO-XXX)
[Metronome](http://themetronome.co/components/XXX/?fresh=true)

**This PR does the following:**
-

### Review Steps:
- [ ] Example step one

### Front End Review Steps:
- [ ] View [Example](http://localhost:3000/)

### CODE REVIEW

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-54oxul4gdvl2gs8qsu1bp6czosvhlwar.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60908b65e1c6005868f07701

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Comment by nypl-wluisi** [comment]
@knezmilos When you get started on this, you'll probably need to run `npm install` after pulling this branch, and then in your local .env file add this `DRUPAL_API=https://sandbox-d8.nypl.org` this will let you use the D8 sandbox environment on AWS to get the content.

**Comment by knezmilos** [comment]
@nypl-wluisi, I've added some styling for the MenuGroup component.  Couple of questions, the main styles will reflect what's in Figma, but the RightRail component's styling is a little different than Figma, for example, links are not underlined, so should the styling be reflected based on the location of the MenuGroup placement on the page?  If it's in `content-secondary` it should be different styling?
Headings on location finder page are a different color, should that be a override on location finder page?

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Comment by nypl-wluisi** [comment]
@knezmilos Lets try adding 2 new props to the MenuGroup component:

`orientation` horizontal or vertical — this would use the link underline or not
`headingColor` hex color value

Also, just a heads up, I had to resolve a merge conflict on this branch, so you'll have to pull this branch back down before making these changes.

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Comment by knezmilos** [comment]
@nypl-wluisi, I've added the `orientation` prop.  I'll need to do some more css cleanup but overall it should work well based on the prop sent.
`headingColor` prop is not added because the `Heading` component in DS doesn't have the "style" property that can be passed.  For now I've just hardcoded the colors in the sidebar and at the bottom.  What do you think if we pass a "span" as the text in the `Heading` component which would have the color passed?
One more thing, based on Figma, sidebar heading looks to be an H2, so I've changed the heading level to be 2 if orientation is vertical.  Does that work?

**Comment by nypl-wluisi** [comment]
@knezmilos I think for now what you have is good. Had a question about the `RightRail` component. I'd like to get rid of this component, and just be able to use the `MenuGroup` component with a basic wrapper for content. Outside of the "content" array, do we actually need any of the styles in `RightRail.scss` file? Can we try to move all the styles into `MenuGroup` ?

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

File: src/pages/locations/index.js [hunk 1]
```diff
@@ -53,15 +54,29 @@ function LocationFinder() {
       contentBottom={
         <Fragment>
           <BottomPromo />
-          <RightRail />
+          <div className="menu-group-horizontal">
```
Comment (nypl-wluisi) [inline src/pages/locations/index.js:57]: @knezmilos can we get rid of this wrapper div and class and have the `orientation` prop handle this instead?

**Comment by nypl-wluisi** [comment]
@knezmilos Can you also update the content for the "Online Resources" sidebar menus to match what's in Figma? https://www.figma.com/file/CCCdFt8H7pMqHjkRQlivSw/IA---Templates-%2B-Components?node-id=41621%3A3611

For ones your don't know the links of, you can just use `https://www.nypl.org`

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Comment by knezmilos** [comment]
@nypl-wluisi, can you take a look at the changes? I was able to remove the wrapper div, but the width will vary slightly on both pages, let me know if you think that's ok. 
I'm going to give a try of changing the whole layout in order to make the sidebarmenu appear above "content bottom" section.  But since the sidebar is using DS's style's, it might be a number of overrides locally to get the layout to match Figma, one being adding styles for another breakpoint for Large Tablet display.

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Comment by nypl-wluisi** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr132-gpyesd6kxdiwntvm2sm8x6xyshypi4lt.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/60a52a40cde72061506af48e

**Pull request merged**